bika/lims/browser/bika_listing.py

```diff
@@ -1514,3 +1575,129 @@ def tabindex(self):
         while True:
             i += 1
             yield i
+
+
+class BikaListingFilterBar(BrowserView):
+    """
+    This class defines a filter bar to make advanced queries in
+    BikaListingView. This filter shouldn't override the 'filter by state'
+    functionality
+    """
+    _render = ViewPageTemplateFile("templates/bika_listing_filter_bar.pt")
+    _filter_bar_dict = {}
+
+    def render(self):
+        """
+        Returns a ViewPageTemplateFile instance with the filter inputs and
+        submit button.
+        """
+        return self._render()
+
+    def setRender(self, new_template):
+        """
+        Defines a new template to render.
+        :new_template: should be a ViewPageTemplateFile object such as
+            'ViewPageTemplateFile("templates/bika_listing_filter_bar.pt")'
+        """
+        if new_template:
+            self._render = new_template
+
+    def filter_bar_button_title(self):
+        """
+        This function returns a string with the name for the input. A function
+        is used in order to translate the name.
+        :return: an string with the title.
+        """
+        return _('Filter')
```

Comment:
This is also an example of where i18n:attributes should be used in TAL, rather than delegating i18n work to the python.  << @Espurna `bika_listing_filter_bar.pt L108`